### ISSUE TITLE
Add the remaining free space

### ISSUE THREAD

**Issue by mohammadrafigh** [Caleb9/downloader#4]
It would be great if you can return the remaining free space from the backend.

**Comment by Caleb9** [comment]
Hi @mohammadrafigh, yeah that could be useful information to display. Implementation is not trivial in .NET though.

In the back-end you have an option to specify the paths where files will be downloaded, e.g. you could set it to `/home/my-user/Downloads/downloader/completed` and `/tmp/downloads/incomplete`. So first thing is which of the two, potentially different, disk volumes space to show (so in our example imagine `/home` and `/tmp` are located on different disks or partitions).

Then, if you want to run the application on non-unix system (so basically on Windows) then the paths would be of different form (`C:\And\So\On`), so back-end would have to take that in consideration too, before reporting back. It's not really a priority for me to run this on Windows but it's something to be aware of.

The last thing I can think of for now is detection of what part of the path actually denotes a mount point for a volume: say you have a disk volume mounted in `/mnt/data-drive/` and you set the option for completed downloads to `/mnt/data-drive/my-stuff/downloads/completed`. Pretty much the only way I found to get available space in .NET is via [DriveInfo](https://learn.microsoft.com/en-us/dotnet/api/System.IO.DriveInfo?view=net-7.0) class. So the back-end would have to somehow deconstruct (in reverse order) elements on the path and match them to mount points to find out which volume the directory is located on. Add to that the possibility of mounting _another_ volume in e.g. `/mnt/data-drive/my-stuff/downloads`, despite `/mnt/data-drive` being a mount point already.

It's certainly doable, but there are some potential pitfalls to be aware of.

**Comment by mohammadrafigh** [comment]
Correct. Maybe just returning all partitions or disk info would be enough?
